src/modules/hip/kernel/jitter.hpp

```diff
@@ -0,0 +1,315 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+#include "rng_seed_stream.hpp"
+
+__device__ void jitter_roi_and_srclocs_hip_compute(int4 *srcRoiPtr_i4, RpptXorwowStateBoxMuller *xorwowState, uint kernelSize, uint bound, int id_x, int id_y, d_float16 *locSrc_f16)
+{
+    d_float8 nhx_f8, nhy_f8;
+    rpp_hip_rng_8_jitter_f32(&nhx_f8, xorwowState);
+    rpp_hip_math_multiply8_const(&nhx_f8, &nhx_f8, (float4)kernelSize);
+    rpp_hip_rng_8_jitter_f32(&nhy_f8, xorwowState);
+    rpp_hip_math_multiply8_const(&nhy_f8, &nhy_f8, (float4)kernelSize);
+
+    d_float8 increment_f8, locDst_f8x, locDst_f8y;
+    increment_f8.f4[0] = make_float4(0.0f, 1.0f, 2.0f, 3.0f);
+    increment_f8.f4[1] = make_float4(4.0f, 5.0f, 6.0f, 7.0f);
+    locDst_f8x.f4[0] = (float4)id_x + increment_f8.f4[0];
```

Comment:
Please use static_cast instead of c style casting

Reply:
Done